### PR TITLE
Fix react router and hot reload and refactor components

### PR DESCRIPTION
**Pull request by chrispicato**

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -1,18 +1,12 @@
 import React, { Component } from 'react';
-import HostLogin from './HostLogin.jsx';
-import Events from './Events.jsx';
 
+// renders logo and maps props
 export default class App extends Component {
-  render() {
+  render () {
```
Comment (tetratorus) [inline client/components/App.jsx]: do we need this space?

**Review by tetratorus — CHANGES_REQUESTED** [review]
hey could you remove the space please? :)

**Review by kevinbrosamle — APPROVED** [review]

**Review by tetratorus — APPROVED** [review]

File: server/server.js [hunk 2]
```diff
@@ -2,11 +2,36 @@
...
 const express = require('express');
 const path = require('path');
-const webpack = require('webpack');
-const WebpackDevServer = require('webpack-dev-server');
-const webpackConfig = require('../webpack.config')
 const config = require('../config');
 const app = express();
+const opn = require('opn');
+
+var webpack = require('webpack'),
```
Comment (andrewk17) [inline server/server.js:9]: var to const

File: server/server.js [hunk 3]
```diff
@@ -2,11 +2,36 @@
...
+    webpackDevMiddleware = require('webpack-dev-middleware'),
+    webpackHotMiddleware = require('webpack-hot-middleware'),
+    webpackconfig = require('../webpack.config.js'),
+    webpackcompiler = webpack(webpackconfig);
+ 
+//enable webpack middleware for hot-reloads in development
+function useWebpackMiddleware(app) {
```
Comment (andrewk17) [inline server/server.js:16]: arrow function

File: server/server.js [hunk 4]
```diff
@@ -2,11 +2,36 @@
...
+        }
+    }));
+    app.use(webpackHotMiddleware(webpackcompiler, {
+        log: console.log
+    }));
+ 
+    return app;
+}
+
+var http = require('http').Server(app);
```
Comment (andrewk17) [inline server/server.js:32]: var to const

File: webpack.config.js [hunk 5]
```diff
@@ -1,13 +1,14 @@
 var path = require('path');
 var webpack = require('webpack');
+var APP_DIR = path.resolve(__dirname, 'client');
+var SERVER_DIR = path.resolve(__dirname, 'server');
 
-
+console.log('=============', path.join(__dirname, 'dist'))
```
Comment (andrewk17) [inline webpack.config.js:6]: do we need this console log

**Review by andrewk17 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**